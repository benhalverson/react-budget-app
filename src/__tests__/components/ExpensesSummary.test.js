import React from 'react';
import moment from 'moment';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ConnectedExpensesSummary, { ExpensesSummary } from '../../components/ExpensesSummary';

const expenses = [
  {
    id: '1',
    description: 'Gum',
    note: '',
    amount: 195,
    createdAt: 0
  },
  {
    id: '2',
    description: 'Rent',
    note: '',
    amount: 109500,
    createdAt: moment(0)
      .subtract(4, 'days')
      .valueOf()
  },
  {
    id: '3',
    description: 'Credit Card',
    note: '',
    amount: 4500,
    createdAt: moment(0)
      .add(4, 'days')
      .valueOf()
  }
];

test('should correctly render expenses summary with one expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235}/>)
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render expenses summary with one expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={1234345553}/>)
  expect(wrapper).toMatchSnapshot();
});

test('should render Expenses summary with props from redux', () => {
  const mockStore = configureStore();
  const store = mockStore({ expenses, filters: { text: '' } });
  const wrapper = shallow(<ConnectedExpensesSummary store={store} />);
  expect(wrapper.props().expenseCount).toEqual(3)
});