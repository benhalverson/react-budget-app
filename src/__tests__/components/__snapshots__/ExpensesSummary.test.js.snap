// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expenses summary with one expense 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenseCount={1}
    expensesTotal={235}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="content-container"
      >
        <h1
          className="page-header__title"
        >
          Viewing 
          <span>
            1
          </span>
           
          expense
           totalling
           
          <span>
            $2.35
          </span>
        </h1>
        <div
          className="page-header__actions"
        >
          <Link
            className="button"
            replace={false}
            to="/create"
          >
            Add Expense
          </Link>
        </div>
      </div>,
      "className": "page-header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="page-header__title"
          >
            Viewing 
            <span>
              1
            </span>
             
            expense
             totalling
             
            <span>
              $2.35
            </span>
          </h1>,
          <div
            className="page-header__actions"
          >
            <Link
              className="button"
              replace={false}
              to="/create"
            >
              Add Expense
            </Link>
          </div>,
        ],
        "className": "content-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Viewing ",
              <span>
                1
              </span>,
              " ",
              "expense",
              " totalling",
              " ",
              <span>
                $2.35
              </span>,
            ],
            "className": "page-header__title",
          },
          "ref": null,
          "rendered": Array [
            "Viewing ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1,
              },
              "ref": null,
              "rendered": 1,
              "type": "span",
            },
            " ",
            "expense",
            " totalling",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "$2.35",
              },
              "ref": null,
              "rendered": "$2.35",
              "type": "span",
            },
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="button"
              replace={false}
              to="/create"
            >
              Add Expense
            </Link>,
            "className": "page-header__actions",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Add Expense",
              "className": "button",
              "replace": false,
              "to": "/create",
            },
            "ref": null,
            "rendered": "Add Expense",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="content-container"
        >
          <h1
            className="page-header__title"
          >
            Viewing 
            <span>
              1
            </span>
             
            expense
             totalling
             
            <span>
              $2.35
            </span>
          </h1>
          <div
            className="page-header__actions"
          >
            <Link
              className="button"
              replace={false}
              to="/create"
            >
              Add Expense
            </Link>
          </div>
        </div>,
        "className": "page-header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="page-header__title"
            >
              Viewing 
              <span>
                1
              </span>
               
              expense
               totalling
               
              <span>
                $2.35
              </span>
            </h1>,
            <div
              className="page-header__actions"
            >
              <Link
                className="button"
                replace={false}
                to="/create"
              >
                Add Expense
              </Link>
            </div>,
          ],
          "className": "content-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Viewing ",
                <span>
                  1
                </span>,
                " ",
                "expense",
                " totalling",
                " ",
                <span>
                  $2.35
                </span>,
              ],
              "className": "page-header__title",
            },
            "ref": null,
            "rendered": Array [
              "Viewing ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 1,
                },
                "ref": null,
                "rendered": 1,
                "type": "span",
              },
              " ",
              "expense",
              " totalling",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "$2.35",
                },
                "ref": null,
                "rendered": "$2.35",
                "type": "span",
              },
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="button"
                replace={false}
                to="/create"
              >
                Add Expense
              </Link>,
              "className": "page-header__actions",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Add Expense",
                "className": "button",
                "replace": false,
                "to": "/create",
              },
              "ref": null,
              "rendered": "Add Expense",
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`should correctly render expenses summary with one expense 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenseCount={23}
    expensesTotal={1234345553}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="content-container"
      >
        <h1
          className="page-header__title"
        >
          Viewing 
          <span>
            23
          </span>
           
          expenses
           totalling
           
          <span>
            $12,343,455.53
          </span>
        </h1>
        <div
          className="page-header__actions"
        >
          <Link
            className="button"
            replace={false}
            to="/create"
          >
            Add Expense
          </Link>
        </div>
      </div>,
      "className": "page-header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="page-header__title"
          >
            Viewing 
            <span>
              23
            </span>
             
            expenses
             totalling
             
            <span>
              $12,343,455.53
            </span>
          </h1>,
          <div
            className="page-header__actions"
          >
            <Link
              className="button"
              replace={false}
              to="/create"
            >
              Add Expense
            </Link>
          </div>,
        ],
        "className": "content-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Viewing ",
              <span>
                23
              </span>,
              " ",
              "expenses",
              " totalling",
              " ",
              <span>
                $12,343,455.53
              </span>,
            ],
            "className": "page-header__title",
          },
          "ref": null,
          "rendered": Array [
            "Viewing ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 23,
              },
              "ref": null,
              "rendered": 23,
              "type": "span",
            },
            " ",
            "expenses",
            " totalling",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "$12,343,455.53",
              },
              "ref": null,
              "rendered": "$12,343,455.53",
              "type": "span",
            },
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="button"
              replace={false}
              to="/create"
            >
              Add Expense
            </Link>,
            "className": "page-header__actions",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Add Expense",
              "className": "button",
              "replace": false,
              "to": "/create",
            },
            "ref": null,
            "rendered": "Add Expense",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="content-container"
        >
          <h1
            className="page-header__title"
          >
            Viewing 
            <span>
              23
            </span>
             
            expenses
             totalling
             
            <span>
              $12,343,455.53
            </span>
          </h1>
          <div
            className="page-header__actions"
          >
            <Link
              className="button"
              replace={false}
              to="/create"
            >
              Add Expense
            </Link>
          </div>
        </div>,
        "className": "page-header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="page-header__title"
            >
              Viewing 
              <span>
                23
              </span>
               
              expenses
               totalling
               
              <span>
                $12,343,455.53
              </span>
            </h1>,
            <div
              className="page-header__actions"
            >
              <Link
                className="button"
                replace={false}
                to="/create"
              >
                Add Expense
              </Link>
            </div>,
          ],
          "className": "content-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Viewing ",
                <span>
                  23
                </span>,
                " ",
                "expenses",
                " totalling",
                " ",
                <span>
                  $12,343,455.53
                </span>,
              ],
              "className": "page-header__title",
            },
            "ref": null,
            "rendered": Array [
              "Viewing ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 23,
                },
                "ref": null,
                "rendered": 23,
                "type": "span",
              },
              " ",
              "expenses",
              " totalling",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "$12,343,455.53",
                },
                "ref": null,
                "rendered": "$12,343,455.53",
                "type": "span",
              },
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="button"
                replace={false}
                to="/create"
              >
                Add Expense
              </Link>,
              "className": "page-header__actions",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Add Expense",
                "className": "button",
                "replace": false,
                "to": "/create",
              },
              "ref": null,
              "rendered": "Add Expense",
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
